In [1]:
import numpy as np
import pandas as pd
import glob

## 분양가 리스트 파일 합치기

In [19]:
# 분양가 리스트 하나의 파일로 합치기

# set the file path
file_path = 'C:/Users/soyou/Dropbox/대체보고서/분양 데이터/apart_price_list/*.csv'

# Get a list of all CSV files in the specified file path
csv_files = glob.glob(file_path)

dataframes = []
for file in csv_files:
    df = pd.read_csv(file)
    dataframes.append(df)

price_df = pd.concat(dataframes)

In [20]:
csv_files

['C:/Users/soyou/Dropbox/대체보고서/분양 데이터/apart_price_list\\apart_list_new_1809-1908_민영.csv',
 'C:/Users/soyou/Dropbox/대체보고서/분양 데이터/apart_price_list\\apart_list_new_1909-2008_민영.csv',
 'C:/Users/soyou/Dropbox/대체보고서/분양 데이터/apart_price_list\\apart_list_new_2009-2108_민영.csv',
 'C:/Users/soyou/Dropbox/대체보고서/분양 데이터/apart_price_list\\apart_list_new_2109-2208_전체.csv',
 'C:/Users/soyou/Dropbox/대체보고서/분양 데이터/apart_price_list\\apart_list_new_2209-2308_전체.csv']

In [21]:
price_df = price_df.reset_index(drop = True)

In [4]:
# 파일저장하기

price_df.to_csv('C:/Users/soyou/Dropbox/대체보고서/분양 데이터/apart_price_list/applied_apt_price_merged.csv', index = False, encoding = 'utf-8-sig')

## 분양 아파트 리스트 파일 합치기

In [6]:
# set the file path
file_path = 'C:/Users/soyou/Dropbox/대체보고서/분양 데이터/apart_list/*.csv'

# Get a list of all CSV files in the specified file path
csv_files = glob.glob(file_path)

dataframes = []
for file in csv_files:
    df = pd.read_csv(file)
    dataframes.append(df)

apt_df = pd.concat(dataframes)

In [7]:
apt_df = apt_df.reset_index(drop = True)
apt_df

지역 주택구분 분양/임대 주택명 \n\t\t\t\t\t\t\t\n순으로 정렬         시공사             문의처  \
0     경기   민영  분양주택         ｅ편한세상　운정　어반프라임Ａ２７（파주     대림산업(주)     ☎ 1600-0922   
1     충남   민영  분양주택           모종　금호어울림　아이퍼스트（아산）     금호산업(주)  ☎ 041-531-0887   
2     경기   민영  분양주택         철산역　롯데캐슬　앤　ＳＫＶＩＥＷ（광명     롯데건설(주)     ☎ 1811-6767   
3     부산   민영  분양주택             남천　더샵　프레스티지　（부산）    (주)포스코건설     ☎ 1661-5155   
4     경북   민영  분양주택                  상주　무양　태왕아너스    (주)태왕이앤씨  ☎ 054-536-8250   
...   ..  ...   ...                          ...         ...             ...   
1962  경남   민영  분양주택               e편한세상 주촌 더프리미어  디엘이앤씨 주식회사  ☎ 055-335-1777   
1963  충북   민영  분양주택                      광혜원 포레가   해성종합건설(주)  ☎ 043-534-8181   
1964  강원   민영  분양주택                  양양 스위트엠 디오션       신태양건설     ☎ 1533-4209   
1965  부산   민영  분양주택                       구서다움파크      비엠산업개발  ☎ 051-514-5533   
1966  충북   민영  분양주택               옥천역 금호어울림 더퍼스트     금호건설(주)  ☎ 043-731-4992   

           모집공고일              청약기간 순으로 정렬 당첨자발표 순으로 정렬 특별공급신청현황 1·2순위 경쟁률  
0     2019-08-30  2019-09-04 ~ 2019-09-06   2019-09-16     신청현황       경쟁률  
1     2019-08-30  2019-09-04 ~ 2019-09-06   2019-09-16     신청현황       경쟁률  
2     2019-08-29  2019-09-04 ~ 2019-09-09   2019-09-17     신청현황       경쟁률  
3     2019-08-29  2019-09-05 ~ 2019-09-10   2019-09-18     신청현황       경쟁률  
4     2019-08-29  2019-09-03 ~ 2019-09-04   2019-09-10   사업주체문의       경쟁률  
...          ...                      ...          ...      ...       ...  
1962  2022-09-02  2022-09-14 ~ 2022-09-16   2022-09-22     신청현황       경쟁률  
1963  2022-09-02  2022-09-13 ~ 2022-09-15   2022-09-21     신청현황       경쟁률  
1964  2022-09-02  2022-09-13 ~ 2022-09-15   2022-09-21     신청현황       경쟁률  
1965  2022-09-02  2022-09-13 ~ 2022-09-16   2022-09-22     신청현황       경쟁률  
1966  2022-09-01  2022-09-06 ~ 2022-09-08   2022-09-19     신청현황       경쟁률  

[1967 rows x 11 columns]

In [13]:
# 컬럼명 수정
apt_df.columns = ['지역', '주택구분', '분양/임대', '아파트이름', '시공사', '문의처',
       '모집공고일', '청약기간 순으로 정렬', '당첨자발표 순으로 정렬', '특별공급신청현황', '1·2순위 경쟁률']

In [30]:
# 파일 저장하기
apt_df.to_csv('C:/Users/soyou/Dropbox/대체보고서/분양 데이터/merge/apt_list_merged.csv', index = False, encoding = 'utf-8-sig')

## 입주예정일 파일 합치기

In [40]:
# set the file path
file_path = 'C:/Users/soyou/Dropbox/대체보고서/분양 데이터/입주예정일/*.csv'

# Get a list of all CSV files in the specified file path
csv_files = glob.glob(file_path)

dataframes = []
for file in csv_files:
    df = pd.read_csv(file)
    dataframes.append(df)

date_df = pd.concat(dataframes)

In [41]:
date_df = date_df.reset_index(drop = True)
date_df

아파트이름    입주예정월
0     ｅ편한세상　운정　어반프라임Ａ２７（파주  2021.07
1       모종　금호어울림　아이퍼스트（아산）  2022.02
2     철산역　롯데캐슬　앤　ＳＫＶＩＥＷ（광명  2022.03
3         남천　더샵　프레스티지　（부산）  2022.09
4              상주　무양　태왕아너스  2021.11
...                    ...      ...
2003        e편한세상 주촌 더프리미어  2025.12
2004               광혜원 포레가  2022.10
2005           양양 스위트엠 디오션  2025.05
2006                구서다움파크  2022.10
2007        옥천역 금호어울림 더퍼스트  2025.01

[2008 rows x 2 columns]

In [26]:
# save the merged file. 
date_df.to_csv('C:/Users/soyou/Dropbox/대체보고서/분양 데이터/merge/date_list_merged.csv', index = False, encoding = 'utf-8-sig')

## 분양가 리스트 파일 기준으로 나머지 파일 merge 하기

### 1. 분양가 리스트와 아파트 리스트(지역, 주택구분, 시공사) merge 하기

In [33]:
# merge에 기준이 되는 분양가 리스트 데이터 shape 확인
price_df.shape

(10643, 5)

In [24]:
# apt_df 에서 '지역', '주택구분' '분양/임대', '시공사' merge

selected_columns = ['아파트이름', '지역', '주택구분', '분양/임대', '시공사']
merge_1 = pd.merge(price_df, apt_df[selected_columns], how = 'left', on = '아파트이름')
merge_1

아파트이름                          공급위치    공급세대       공급면적  \
0      ｅ편한세상　운정　어반프라임Ａ２７（파주     경기도　파주시　운정３지구　Ａ－２７블록（동패동）  1009세대  059.6242A   
1      ｅ편한세상　운정　어반프라임Ａ２７（파주     경기도　파주시　운정３지구　Ａ－２７블록（동패동）  1009세대  059.3403B   
2      ｅ편한세상　운정　어반프라임Ａ２７（파주     경기도　파주시　운정３지구　Ａ－２７블록（동패동）  1009세대  059.9747C   
3      ｅ편한세상　운정　어반프라임Ａ２７（파주     경기도　파주시　운정３지구　Ａ－２７블록（동패동）  1009세대  074.7261A   
4      ｅ편한세상　운정　어반프라임Ａ２７（파주     경기도　파주시　운정３지구　Ａ－２７블록（동패동）  1009세대  084.8864A   
...                     ...                           ...     ...        ...   
10797                구서다움파크  부산광역시 금정구 중앙대로1893번길 50(구서동)    84세대   076.9510   
10798        옥천역 금호어울림 더퍼스트   충청북도 옥천군 옥천읍 마암리 산 2-3번지 일원   499세대  084.9114A   
10799        옥천역 금호어울림 더퍼스트   충청북도 옥천군 옥천읍 마암리 산 2-3번지 일원   499세대  084.9566B   
10800        옥천역 금호어울림 더퍼스트   충청북도 옥천군 옥천읍 마암리 산 2-3번지 일원   499세대  084.9118C   
10801        옥천역 금호어울림 더퍼스트   충청북도 옥천군 옥천읍 마암리 산 2-3번지 일원   499세대   114.9688   

         공급금액  지역 주택구분 분양/임대      시공사  
0      30,960  경기   민영  분양주택  대림산업(주)  
1      30,960  경기   민영  분양주택  대림산업(주)  
2      28,060  경기   민영  분양주택  대림산업(주)  
3      36,950  경기   민영  분양주택  대림산업(주)  
4      41,390  경기   민영  분양주택  대림산업(주)  
...       ...  ..  ...   ...      ...  
10797  59,090  부산   민영  분양주택   비엠산업개발  
10798  32,970  충북   민영  분양주택  금호건설(주)  
10799  33,170  충북   민영  분양주택  금호건설(주)  
10800  33,120  충북   민영  분양주택  금호건설(주)  
10801  49,050  충북   민영  분양주택  금호건설(주)  

[10802 rows x 9 columns]

=> price_df 기준으로 merge 하면 rows 수가 10643이어야 하는데 10802임

=> apt_df에 중복 있는지 확인 필요

In [65]:
# apt_df에 중복 있음. 
duplicated_rows = apt_df[apt_df['아파트이름'].duplicated(keep=False)]['아파트이름']
print(duplicated_rows)

176     수원광교４０Ａ２６　８５초과　공임예비
371     수원광교４０Ａ２６　８５초과　공임예비
484            쌍용 더 플래티넘 광주
524                   골든렉시움
536         이안 센트럴포레 장유 2단지
537         이안 센트럴포레 장유 1단지
560            나운 금호어울림 센트럴
634         파주연풍 양우내안애 에코하임
891          마전 양우 내안애 퍼스트힐
922        더샵 지제역 센트럴파크 1BL
1089        파주연풍 양우내안애 에코하임
1273           나운 금호어울림 센트럴
1303              시흥 센트럴 헤센
1324         마전 양우 내안애 퍼스트힐
1330     울산 온양발리 신일해피트리 더루츠
1340       더샵 지제역 센트럴파크 1BL
1576        이안 센트럴포레 장유 2단지
1577        이안 센트럴포레 장유 1단지
1586           쌍용 더 플래티넘 광주
1590                  골든렉시움
1710        이안 센트럴포레 장유 2단지
1711        이안 센트럴포레 장유 1단지
1753              시흥 센트럴 헤센
1778     울산 온양발리 신일해피트리 더루츠
1793              군산 한성필하우스
1899        파주연풍 양우내안애 에코하임
1952              군산 한성필하우스
Name: 아파트이름, dtype: object


=> 확인결과 입주자 모집을 두 번 공고낸 케이스들.

=> 일단 삭제

In [66]:
# apt_df의 아파트명 중복 되는 것 삭제

apt_df.drop_duplicates(subset=['아파트이름'], ignore_index = True, keep= False, inplace = True)

In [67]:
# 중복 있는지 다시 확인
duplicated_rows_2 = apt_df[apt_df['아파트이름'].duplicated(keep=False)]['아파트이름']
print(duplicated_rows_2)
# => 없음

Series([], Name: 아파트이름, dtype: object)


In [81]:
# price_df에도 해당 아파트 명 삭제  (총 122 row 삭제)

duplicated_list = list(set(duplicated_rows))

price_df_drop = price_df.drop(price_df.index[price_df['아파트이름'].isin(duplicated_list)]).reset_index(drop = True)
price_df_drop.shape

(10521, 5)

In [82]:
# 다시 merge하기
selected_columns = ['아파트이름', '지역', '주택구분', '분양/임대', '시공사']
merge_1 = pd.merge(price_df_drop, apt_df[selected_columns], how = 'left', on = '아파트이름')
merge_1

아파트이름                          공급위치    공급세대       공급면적  \
0      ｅ편한세상　운정　어반프라임Ａ２７（파주     경기도　파주시　운정３지구　Ａ－２７블록（동패동）  1009세대  059.6242A   
1      ｅ편한세상　운정　어반프라임Ａ２７（파주     경기도　파주시　운정３지구　Ａ－２７블록（동패동）  1009세대  059.3403B   
2      ｅ편한세상　운정　어반프라임Ａ２７（파주     경기도　파주시　운정３지구　Ａ－２７블록（동패동）  1009세대  059.9747C   
3      ｅ편한세상　운정　어반프라임Ａ２７（파주     경기도　파주시　운정３지구　Ａ－２７블록（동패동）  1009세대  074.7261A   
4      ｅ편한세상　운정　어반프라임Ａ２７（파주     경기도　파주시　운정３지구　Ａ－２７블록（동패동）  1009세대  084.8864A   
...                     ...                           ...     ...        ...   
10516                구서다움파크  부산광역시 금정구 중앙대로1893번길 50(구서동)    84세대   076.9510   
10517        옥천역 금호어울림 더퍼스트   충청북도 옥천군 옥천읍 마암리 산 2-3번지 일원   499세대  084.9114A   
10518        옥천역 금호어울림 더퍼스트   충청북도 옥천군 옥천읍 마암리 산 2-3번지 일원   499세대  084.9566B   
10519        옥천역 금호어울림 더퍼스트   충청북도 옥천군 옥천읍 마암리 산 2-3번지 일원   499세대  084.9118C   
10520        옥천역 금호어울림 더퍼스트   충청북도 옥천군 옥천읍 마암리 산 2-3번지 일원   499세대   114.9688   

         공급금액  지역 주택구분 분양/임대      시공사  
0      30,960  경기   민영  분양주택  대림산업(주)  
1      30,960  경기   민영  분양주택  대림산업(주)  
2      28,060  경기   민영  분양주택  대림산업(주)  
3      36,950  경기   민영  분양주택  대림산업(주)  
4      41,390  경기   민영  분양주택  대림산업(주)  
...       ...  ..  ...   ...      ...  
10516  59,090  부산   민영  분양주택   비엠산업개발  
10517  32,970  충북   민영  분양주택  금호건설(주)  
10518  33,170  충북   민영  분양주택  금호건설(주)  
10519  33,120  충북   민영  분양주택  금호건설(주)  
10520  49,050  충북   민영  분양주택  금호건설(주)  

[10521 rows x 9 columns]

=> rows 수 10521으로 merge 잘 됨

### 2. merge_1에 입주예정월 (date_df) 합치기

In [95]:
duplicated_list

['울산 온양발리 신일해피트리 더루츠',
 '마전 양우 내안애 퍼스트힐',
 '이안 센트럴포레 장유 1단지',
 '수원광교４０Ａ２６\u3000８５초과\u3000공임예비',
 '쌍용 더 플래티넘 광주',
 '골든렉시움',
 '더샵 지제역 센트럴파크 1BL',
 '파주연풍 양우내안애 에코하임',
 '시흥 센트럴 헤센',
 '나운 금호어울림 센트럴',
 '군산 한성필하우스',
 '이안 센트럴포레 장유 2단지']

In [91]:
# 앞서 확인한 데이터들의 중복리스트 제거하기

date_df_drop = date_df.drop(date_df.index[date_df['아파트이름'].isin(duplicated_list)]).reset_index(drop = True)
date_df_drop.shape

(1979, 2)

In [92]:
# 추가적인 중복 row 확인하기

duplicated = date_df_drop[date_df_drop['아파트이름'].duplicated(keep=False)]['아파트이름']
print(duplicated)

471       가화만사성 정관타운
949       가화만사성 정관타운
1006    양산 덕계 경동스마트홈
1209    양산 덕계 경동스마트홈
Name: 아파트이름, dtype: object


In [103]:
# merge1에 있는 아파트인지 확인 => 양산 덕계는 없음.. 가화만사성은 

apt_df.loc[(apt_df['아파트이름'] == '가화만사성 정관타운') | (apt_df['아파트이름'] == '양산 덕계 경동스마트홈')]

지역 주택구분 분양/임대       아파트이름      시공사             문의처       모집공고일  \
471  부산   민영  분양주택  가화만사성 정관타운  (주)가화건설  ☎ 051-724-5111  2020-07-08   

                 청약기간 순으로 정렬 당첨자발표 순으로 정렬 특별공급신청현황 1·2순위 경쟁률  
471  2020-07-21 ~ 2020-07-23   2020-07-29     신청현황       경쟁률

In [105]:
date_df_drop.loc[date_df_drop['아파트이름'] == '가화만사성 정관타운']

아파트이름    입주예정월
471  가화만사성 정관타운  2020.08
949  가화만사성 정관타운  2021.06

In [111]:
# 가화만사성 정관타운 2021.06 삭제 (홈페이지 확인)

date_df_drop_f = date_df_drop.drop(index = 949)
date_df_drop_f.shape

(1978, 2)

In [112]:
# merge_1에 입주예정월 merge

merge_2 = pd.merge(merge_1, date_df_drop_f, how = 'left', on = '아파트이름')
merge_2

아파트이름                          공급위치    공급세대       공급면적  \
0      ｅ편한세상　운정　어반프라임Ａ２７（파주     경기도　파주시　운정３지구　Ａ－２７블록（동패동）  1009세대  059.6242A   
1      ｅ편한세상　운정　어반프라임Ａ２７（파주     경기도　파주시　운정３지구　Ａ－２７블록（동패동）  1009세대  059.3403B   
2      ｅ편한세상　운정　어반프라임Ａ２７（파주     경기도　파주시　운정３지구　Ａ－２７블록（동패동）  1009세대  059.9747C   
3      ｅ편한세상　운정　어반프라임Ａ２７（파주     경기도　파주시　운정３지구　Ａ－２７블록（동패동）  1009세대  074.7261A   
4      ｅ편한세상　운정　어반프라임Ａ２７（파주     경기도　파주시　운정３지구　Ａ－２７블록（동패동）  1009세대  084.8864A   
...                     ...                           ...     ...        ...   
10516                구서다움파크  부산광역시 금정구 중앙대로1893번길 50(구서동)    84세대   076.9510   
10517        옥천역 금호어울림 더퍼스트   충청북도 옥천군 옥천읍 마암리 산 2-3번지 일원   499세대  084.9114A   
10518        옥천역 금호어울림 더퍼스트   충청북도 옥천군 옥천읍 마암리 산 2-3번지 일원   499세대  084.9566B   
10519        옥천역 금호어울림 더퍼스트   충청북도 옥천군 옥천읍 마암리 산 2-3번지 일원   499세대  084.9118C   
10520        옥천역 금호어울림 더퍼스트   충청북도 옥천군 옥천읍 마암리 산 2-3번지 일원   499세대   114.9688   

         공급금액  지역 주택구분 분양/임대      시공사    입주예정월  
0      30,960  경기   민영  분양주택  대림산업(주)  2021.07  
1      30,960  경기   민영  분양주택  대림산업(주)  2021.07  
2      28,060  경기   민영  분양주택  대림산업(주)  2021.07  
3      36,950  경기   민영  분양주택  대림산업(주)  2021.07  
4      41,390  경기   민영  분양주택  대림산업(주)  2021.07  
...       ...  ..  ...   ...      ...      ...  
10516  59,090  부산   민영  분양주택   비엠산업개발  2022.10  
10517  32,970  충북   민영  분양주택  금호건설(주)  2025.01  
10518  33,170  충북   민영  분양주택  금호건설(주)  2025.01  
10519  33,120  충북   민영  분양주택  금호건설(주)  2025.01  
10520  49,050  충북   민영  분양주택  금호건설(주)  2025.01  

[10521 rows x 10 columns]

In [113]:
# NA 확인하기
merge_2.isna().sum()

아파트이름     0
공급위치      0
공급세대      0
공급면적      0
공급금액      0
지역       13
주택구분     13
분양/임대    13
시공사      51
입주예정월     0
dtype: int64

In [117]:
merge_2.loc[merge_2['시공사'].isna(), :]

아파트이름                                   공급위치  \
2522               디에이치 퍼스티어 아이파크              서울특별시 강남구 개포1동 660-4번지 일원   
2523               디에이치 퍼스티어 아이파크              서울특별시 강남구 개포1동 660-4번지 일원   
2524               디에이치 퍼스티어 아이파크              서울특별시 강남구 개포1동 660-4번지 일원   
2525               디에이치 퍼스티어 아이파크              서울특별시 강남구 개포1동 660-4번지 일원   
2526               디에이치 퍼스티어 아이파크              서울특별시 강남구 개포1동 660-4번지 일원   
2527               디에이치 퍼스티어 아이파크              서울특별시 강남구 개포1동 660-4번지 일원   
2528               디에이치 퍼스티어 아이파크              서울특별시 강남구 개포1동 660-4번지 일원   
2529               디에이치 퍼스티어 아이파크              서울특별시 강남구 개포1동 660-4번지 일원   
2530               디에이치 퍼스티어 아이파크              서울특별시 강남구 개포1동 660-4번지 일원   
2531               디에이치 퍼스티어 아이파크              서울특별시 강남구 개포1동 660-4번지 일원   
2532               디에이치 퍼스티어 아이파크              서울특별시 강남구 개포1동 660-4번지 일원   
2781               여주역 금호어울림 베르티스                경기도 여주시 교동로 171-24 (교동)   
2782               여주역 금호어울림 베르티스                경기도 여주시 교동로 171-24 (교동)   
2783               여주역 금호어울림 베르티스                경기도 여주시 교동로 171-24 (교동)   
2784               여주역 금호어울림 베르티스                경기도 여주시 교동로 171-24 (교동)   
2947                  남천 우성스마트시티뷰            부산광역시 수영구 황령대로509번길 9 (남천동)   
2948                  남천 우성스마트시티뷰            부산광역시 수영구 황령대로509번길 9 (남천동)   
2949                  남천 우성스마트시티뷰            부산광역시 수영구 황령대로509번길 9 (남천동)   
2950                  남천 우성스마트시티뷰            부산광역시 수영구 황령대로509번길 9 (남천동)   
2999               이트리니티 공도 센트럴파크                  경기도 안성시 공도읍 도화길 67-14   
3000               이트리니티 공도 센트럴파크                  경기도 안성시 공도읍 도화길 67-14   
3001               이트리니티 공도 센트럴파크                  경기도 안성시 공도읍 도화길 67-14   
3002               이트리니티 공도 센트럴파크                  경기도 안성시 공도읍 도화길 67-14   
3003               이트리니티 공도 센트럴파크                  경기도 안성시 공도읍 도화길 67-14   
3004               이트리니티 공도 센트럴파크                  경기도 안성시 공도읍 도화길 67-14   
3005               이트리니티 공도 센트럴파크                  경기도 안성시 공도읍 도화길 67-14   
3006               이트리니티 공도 센트럴파크                  경기도 안성시 공도읍 도화길 67-14   
3018  제38차 장기전세주택 입주자 모집 공고(85초과)  서울특별시 강동구 고덕로 427 (강일동, 고덕리엔파크2단지아파트)   
3181               의정부역 진산엔월드메르디앙                    경기도 의정부시 의정부동 378-1   
3182               의정부역 진산엔월드메르디앙                    경기도 의정부시 의정부동 378-1   
3183               의정부역 진산엔월드메르디앙                    경기도 의정부시 의정부동 378-1   
3184               의정부역 진산엔월드메르디앙                    경기도 의정부시 의정부동 378-1   
3227                      호반써밋 목동            서울특별시 양천구 신정3동 1150-41번지 일원   
3228                      호반써밋 목동            서울특별시 양천구 신정3동 1150-41번지 일원   
3229                      호반써밋 목동            서울특별시 양천구 신정3동 1150-41번지 일원   
3230                      호반써밋 목동            서울특별시 양천구 신정3동 1150-41번지 일원   
3231                      호반써밋 목동            서울특별시 양천구 신정3동 1150-41번지 일원   
3232                      호반써밋 목동            서울특별시 양천구 신정3동 1150-41번지 일원   
5830   고덕신도시 EBC-2BL 힐스테이트 고덕 센트럴             경기도 평택시 고덕면 고덕국제화지구EBC-2BL   
5831   고덕신도시 EBC-2BL 힐스테이트 고덕 센트럴             경기도 평택시 고덕면 고덕국제화지구EBC-2BL   
5832   고덕신도시 EBC-2BL 힐스테이트 고덕 센트럴             경기도 평택시 고덕면 고덕국제화지구EBC-2BL   
5833   고덕신도시 EBC-2BL 힐스테이트 고덕 센트럴             경기도 평택시 고덕면 고덕국제화지구EBC-2BL   
5834   고덕신도시 EBC-2BL 힐스테이트 고덕 센트럴             경기도 평택시 고덕면 고덕국제화지구EBC-2BL   
5835   고덕신도시 EBC-2BL 힐스테이트 고덕 센트럴             경기도 평택시 고덕면 고덕국제화지구EBC-2BL   
5836   고덕신도시 EBC-2BL 힐스테이트 고덕 센트럴             경기도 평택시 고덕면 고덕국제화지구EBC-2BL   
7318              센텀 아스트룸 SK VIEW               부산광역시 해운대구 반여동 1349번지 일원   
7319              센텀 아스트룸 SK VIEW               부산광역시 해운대구 반여동 1349번지 일원   
7320              센텀 아스트룸 SK VIEW               부산광역시 해운대구 반여동 1349번지 일원   
7321              센텀 아스트룸 SK VIEW               부산광역시 해운대구 반여동 1349번지 일원   
7322              센텀 아스트룸 SK VIEW               부산광역시 해운대구 반여동 1349번지 일원   
7323              센텀 아스트룸 SK VIEW               부산광역시 해운대구 반여동 1349번지 일원   

        공급세대       공급면적     공급금액   지역 주택구분      분양/임대  시공사    입주예정월  
2522 

In [121]:
na_list= merge_2.loc[merge_2['시공사'].isna(), :]['아파트이름'].unique().tolist()
na_list

['디에이치 퍼스티어 아이파크',
 '여주역 금호어울림 베르티스',
 '남천 우성스마트시티뷰',
 '이트리니티 공도 센트럴파크',
 '제38차 장기전세주택 입주자 모집 공고(85초과)',
 '의정부역 진산엔월드메르디앙',
 '호반써밋 목동',
 '고덕신도시 EBC-2BL 힐스테이트 고덕 센트럴',
 '센텀 아스트룸 SK VIEW']

In [97]:
apt_df_2 = pd.read_csv('C:/Users/soyou/Dropbox/대체보고서/분양 데이터/merge/apt_list_merged.csv')

In [124]:
apt_df_2.loc[apt_df_2['아파트이름'].isin(na_list),:]

지역 주택구분      분양/임대                        아파트이름  시공사             문의처  \
472  서울   민영       분양주택               디에이치 퍼스티어 아이파크  NaN  ☎ 02-2058-2153   
523  경기   민영       분양주택               여주역 금호어울림 베르티스  NaN     ☎ 1566-5755   
553  부산   민영       분양주택                  남천 우성스마트시티뷰  NaN  ☎ 051-528-2229   
564  경기   민영       분양주택               이트리니티 공도 센트럴파크  NaN  ☎ 031-658-7520   
567  서울   민영  분양전환 불가임대  제38차 장기전세주택 입주자 모집 공고(85초과)  NaN     ☎ 1600-3456   
594  경기   민영       분양주택               의정부역 진산엔월드메르디앙  NaN  ☎ 031-836-7700   
604  서울   민영       분양주택                      호반써밋 목동  NaN     ☎ 1566-3382   

          모집공고일              청약기간 순으로 정렬 당첨자발표 순으로 정렬 특별공급신청현황 1·2순위 경쟁률  
472  2020-07-09  2020-07-20 ~ 2020-07-23   2020-07-29     신청현황       경쟁률  
523  2020-06-05  2020-06-17 ~ 2020-06-19   2020-06-25     신청현황       경쟁률  
553  2020-05-22  2020-06-01 ~ 2020-06-03   2020-06-09     신청현황       경쟁률  
564  2020-05-15  2020-05-25 ~ 2020-05-27   2020-06-02     신청현황       경쟁률  
567  2020-05-15  2020-05-28 ~ 2020-06-01   2020-09-25   사업주체문의    사업주체문의  
594  2020-04-24  2020-05-04 ~ 2020-05-06   2020-05-12   사업주체문의       경쟁률  
604  2020-04-10  2020-04-20 ~ 2020-04-23   2020-04-29     신청현황       경쟁률

=> 시공사만 없는 경우는 애초부터 공란이었기 때문. 상세보기에는 시행사가 '00조합'으로 들어가 있는데, 여러 건설사가 참여하는 경우임.

=> 지역, 주택구분, 분양/임대, 시공사가 누락된 고덕센트럴과 센텀 SK VIEW는 수기로 채워넣기

In [131]:
# 누락된 정보 수기로 채워넣기

# 고덕신도시 EBC-2BL 힐스테이트 고덕 센트럴
# 시공사 : 현대엔지니어링 주식회사
# 지역 : 경기
# 주택구분 : 민영
# 분양/임대 : 분양주택

# 센텀 아스트룸 SK VIEW
# 시공사 : SK에코플랜트(주)
# 지역 : 부산
# 주택구분 : 민영
# 분양/임대 : 분양주택

merge_2.loc[merge_2['아파트이름'] == '고덕신도시 EBC-2BL 힐스테이트 고덕 센트럴', '지역'] = '경기'
merge_2.loc[merge_2['아파트이름'] == '고덕신도시 EBC-2BL 힐스테이트 고덕 센트럴', '주택구분'] = '민영'
merge_2.loc[merge_2['아파트이름'] == '고덕신도시 EBC-2BL 힐스테이트 고덕 센트럴', '분양/임대'] = '분양주택'
merge_2.loc[merge_2['아파트이름'] == '고덕신도시 EBC-2BL 힐스테이트 고덕 센트럴', '시공사'] = '현대엔지니어링 주식회사'

merge_2.loc[merge_2['아파트이름'] == '센텀 아스트룸 SK VIEW', '지역'] = '부산'
merge_2.loc[merge_2['아파트이름'] == '센텀 아스트룸 SK VIEW', '주택구분'] = '민영'
merge_2.loc[merge_2['아파트이름'] == '센텀 아스트룸 SK VIEW', '분양/임대'] = '분양주택'
merge_2.loc[merge_2['아파트이름'] == '센텀 아스트룸 SK VIEW', '시공사'] = 'SK에코플랜트(주)'

In [134]:
# 잘 채워졌는지 확인 

merge_2.loc[(merge_2['아파트이름'] == '고덕신도시 EBC-2BL 힐스테이트 고덕 센트럴') | (merge_2['아파트이름'] == '센텀 아스트룸 SK VIEW')]

아파트이름                        공급위치   공급세대  \
5830  고덕신도시 EBC-2BL 힐스테이트 고덕 센트럴  경기도 평택시 고덕면 고덕국제화지구EBC-2BL  660세대   
5831  고덕신도시 EBC-2BL 힐스테이트 고덕 센트럴  경기도 평택시 고덕면 고덕국제화지구EBC-2BL  660세대   
5832  고덕신도시 EBC-2BL 힐스테이트 고덕 센트럴  경기도 평택시 고덕면 고덕국제화지구EBC-2BL  660세대   
5833  고덕신도시 EBC-2BL 힐스테이트 고덕 센트럴  경기도 평택시 고덕면 고덕국제화지구EBC-2BL  660세대   
5834  고덕신도시 EBC-2BL 힐스테이트 고덕 센트럴  경기도 평택시 고덕면 고덕국제화지구EBC-2BL  660세대   
5835  고덕신도시 EBC-2BL 힐스테이트 고덕 센트럴  경기도 평택시 고덕면 고덕국제화지구EBC-2BL  660세대   
5836  고덕신도시 EBC-2BL 힐스테이트 고덕 센트럴  경기도 평택시 고덕면 고덕국제화지구EBC-2BL  660세대   
7318             센텀 아스트룸 SK VIEW    부산광역시 해운대구 반여동 1349번지 일원  544세대   
7319             센텀 아스트룸 SK VIEW    부산광역시 해운대구 반여동 1349번지 일원  544세대   
7320             센텀 아스트룸 SK VIEW    부산광역시 해운대구 반여동 1349번지 일원  544세대   
7321             센텀 아스트룸 SK VIEW    부산광역시 해운대구 반여동 1349번지 일원  544세대   
7322             센텀 아스트룸 SK VIEW    부산광역시 해운대구 반여동 1349번지 일원  544세대   
7323             센텀 아스트룸 SK VIEW    부산광역시 해운대구 반여동 1349번지 일원  544세대   

           공급면적    공급금액  지역 주택구분 분양/임대           시공사    입주예정월  
5830   078.1643  49,200  경기   민영  분양주택  현대엔지니어링 주식회사  2024.08  
5831  084.3909A  53,100  경기   민영  분양주택  현대엔지니어링 주식회사  2024.08  
5832  084.3389B  53,600  경기   민영  분양주택  현대엔지니어링 주식회사  2024.08  
5833  093.1000A  58,800  경기   민영  분양주택  현대엔지니어링 주식회사  2024.08  
5834  093.1175B  58,800  경기   민영  분양주택  현대엔지니어링 주식회사  2024.08  
5835  093.8620C  59,500  경기   민영  분양주택  현대엔지니어링 주식회사  2024.08  
5836  093.9621D  59,500  경기   민영  분양주택  현대엔지니어링 주식회사  2024.08  
7318  059.9930A  45,300  부산   민영  분양주택    SK에코플랜트(주)  2024.06  
7319  059.9761B  44,800  부산   민영  분양주택    SK에코플랜트(주)  2024.06  
7320  074.7150A  57,200  부산   민영  분양주택    SK에코플랜트(주)  2024.06  
7321  074.7417B  56,700  부산   민영  분양주택    SK에코플랜트(주)  2024.06  
7322  084.9807A  66,100  부산   민영  분양주택    SK에코플랜트(주)  2024.06  
7323  084.9853B  64,900  부산   민영  분양주택    SK에코플랜트(주)  2024.06

In [135]:
# NA 최종확인 
merge_2.isna().sum()

아파트이름     0
공급위치      0
공급세대      0
공급면적      0
공급금액      0
지역        0
주택구분      0
분양/임대     0
시공사      38
입주예정월     0
dtype: int64

In [137]:
# 파일저장하기

merge_2.to_csv('C:/Users/soyou/Dropbox/대체보고서/분양 데이터/merge/final_merged_apt_data.csv', index = False, encoding = 'utf-8-sig')